addons/spreadsheet/static/src/chart/odoo_chart/odoo_pie_chart.js

```diff
@@ -21,6 +21,10 @@ chartRegistry.add("odoo_pie", {
 
 function createOdooChartRuntime(chart, getters) {
     const background = chart.background || "#FFFFFF";
+    if (chart.dataSource.getData() === null) {
+       const chartJsConfig = getPieConfiguration(chart, []);
+       return { background, chartJsConfig };
+    }
```

Comment:
It would be more robust to handle this upstream, in the data source itself.
There's several other occurrence of  `const { datasets, labels } = chart.dataSource.getData();` where the issue could also potentially arise.

I suggest the following change

```patch
diff --git a/addons/spreadsheet/static/src/chart/data_source/chart_data_source.js b/addons/spreadsheet/static/src/chart/data_source/chart_data_source.js
index 7a3fe89c0a1..705ab337e69 100644
--- a/addons/spreadsheet/static/src/chart/data_source/chart_data_source.js
+++ b/addons/spreadsheet/static/src/chart/data_source/chart_data_source.js
@@ -42,6 +42,6 @@ export default class ChartDataSource extends OdooViewsDataSource {
         if (!this._isValid) {
             return { datasets: [], labels: [] };
         }
-        return this._model.data;
+        return this._model.data || { datasets: [], labels: [] };
     }
 }

```